web/src/server_events_dispatch.js

```diff
@@ -253,6 +254,7 @@ export function dispatch_normal_event(event) {
             switch (event.op) {
                 case "add":
                     reactions.add_reaction(event);
+                    reaction_notifications.received_reaction(event);
```

Comment:
If we don't have the message in cache, `add_reaction` simply returns. But `received_reaction` fetches the message and sends notification - have we discussed it somewhere to make this change in behaviour ?

Also, what if someone "liked message" -> user received desktop notification -> "unliked message"; we need to revoke notification in this case. See `desktop_notifications.close_notification`